swc.config.ts

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable @typescript-eslint/no-require-imports */
+const { swcDir } = require("@swc/cli");
+import type { Options } from "@swc/core";
+import path from "path";
+
+/**
+ * This script configures SWC programatically to compile the project.
+ *
+ * The main reason for doing this over using .swcrc is to be able to
+ * dynamically set feature flags based on environment variables. These
+ * feature flags can then be used in the code to conditionally compile
+ * different code paths.
+ *
+ * These are not neccessarily user facing features, they are more like
+ * infrastructure within the package. An example could be to enable
+ * debug logging in development, or to enable fetch with credentials
+ * during development.
+ *
+ * Most of this code was generated in conversation with Uncle Claude,
+ * in case you were wondering about unusual conventions...
+ *
+ */
+
+// Define all possible feature flags. They key will be turned into a
+// global variable wrapped with double underscores, e.g. __DEBUG_MODE__,
+// and the value will be the environment variable to read.
+const COMPILE_FLAGS = {
+  ENABLE_FETCH_WITH_CREDENTIALS: "ENABLE_FETCH_WITH_CREDENTIALS",
+  // DEBUG_MODE: 'DEBUG_MODE',
+} as const;
+
+// Helper to convert environment variable to boolean
+const envToBool = (value: string | undefined): boolean => {
+  if (!value) return false;
+  return ["1", "true", "yes"].includes(value.toLowerCase());
+};
+
+// Get all feature flags from environment
+const getFeatureFlags = () => {
```

Comment:
Looking at what's in the `voltaire` PR - would it not make sense to move and export this (also renaming to `getCompileFlags` to reflect the change to `COMPILE_FLAGS`) and allow the flags as an argument, so that we can de-duplicate `envToBool` and `getFeatureFlags` in that PR?

Reply:
@jamiehenson I worry that it creates a strong coupling between two different compiler architectures and respective plugins. Its 10 lines, I'm okay with it being duplicated 😇. I will rename though, apologies for doing a rush job